bugsnag-plugin-android-ndk/src/main/jni/utils/serializer.c

```diff
@@ -694,20 +694,32 @@ void bsg_serialize_error(bsg_error exc, JSON_Object *exception,
   json_object_set_string(exception, "errorClass", exc.errorClass);
   json_object_set_string(exception, "message", exc.errorMessage);
   json_object_set_string(exception, "type", "c");
-  for (int findex = 0; findex < exc.frame_count; findex++) {
+  // assuming that the initial frame is the program counter. This logic will
+  // need to be revisited if (for example) we add more intelligent processing
+  // for stack overflow-type errors, like discarding the top frames, which
+  // would mean no stored frame is the program counter.
+  if (exc.frame_count > 0) {
+    bsg_serialize_stackframe(&(exc.stacktrace[0]), true, stacktrace);
+  }
+  for (int findex = 1; findex < exc.frame_count; findex++) {
```

Comment:
nit: this for loop could be simplified by using `bool isPc = findex == 0`

Reply:
Correct, and the initial implementation worked that way, though this opted for perf over doing the index check every loop